lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -325,6 +327,15 @@ class ProviderTest {
         assertNotNull(response)
     }
 
+    @ParameterizedTest
+    @MethodSource("getProviders")
+    fun `get transaction receipt throws on incorrect hash`(provider: Provider) {
+        val request = provider.getTransactionReceipt(Felt.ZERO)
+        assertThrows(RequestFailedException::class.java) {
+            request.send()
+        }
```

Comment:
This is problematic, as the GatewayTransactionReceipt is defined in such a way that querying the network for unknown hashes still works. However RPC provider returns RequestFailedException.

This is inconsistent (that's why this test currently fails) and I wonder if it should be kept that way?

Reply:
Well I think the RPC error message actually indicates what happened, but I'll verify.

Reply:
I'll add a separate issue to handle this. Tried resolving currently, but https://github.com/Shard-Labs/starknet-devnet/issues/366 is blocking my current approach from working with local tests and don't want to do workarounds.